homeassistant/components/bthome/binary_sensor.py

```diff
@@ -195,3 +196,15 @@ class BTHomeBluetoothBinarySensorEntity(
     def is_on(self) -> bool | None:
         """Return the native value."""
         return self.processor.entity_data.get(self.entity_key)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        if self.device_info:
+            if self.device_info[ATTR_MODEL] and self.device_info[ATTR_MODEL].endswith(
+                "(trigger based device)"
+            ):
```

Comment:
Ideally we would avoid string matching here and set a property on the processor object to make it as a trigger based device.

Reply:
Should be resolved now.